modules/sdk-coin-sui/src/lib/transaction.ts

```diff
@@ -1,16 +1,354 @@
+   * Load the input and output data on this transaction.
+   */
+  loadInputsAndOutputs(): void {
+    if (!this.suiTransaction) {
+      return;
+    }
+
+    const tx = this.suiTransaction;
+    const payTx = tx.kind.Single.Pay;
+    const recipients = payTx.recipients;
+    const amounts = payTx.amounts;
+    if (recipients.length !== amounts.length) {
+      throw new Error('The length of recipients does not equal to the length of amounts');
+    }
+
+    const outputs: Entry[] = recipients.map((recipient, index) => ({
+      address: recipient,
+      value: amounts[index].toString(),
+      coin: this._coinConfig.name,
+    }));
+
+    const totalAmount = amounts.reduce((accumulator, current) => accumulator + current, 0);
+    const inputs: Entry[] = payTx.coins.map((input) => ({
+      address: tx.sender,
+      value: totalAmount.toString(),
+      coin: this._coinConfig.name,
+    }));
+
+    this._inputs = inputs;
+    this._outputs = outputs;
+  }
+
+  /**
+   * Sets this transaction payload
+   *
+   * @param rawTransaction
+   */
+  fromRawTransaction(rawTransaction: string): void {
+    try {
+      utils.isValidRawTransaction(rawTransaction);
+      this._suiTransaction = Transaction.deserializeSuiTransaction(rawTransaction);
+      this._type = TransactionType.Send;
+      this.loadInputsAndOutputs();
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  serialize(): string {
+    const suiTx = this._suiTransaction;
+    const txData = {
+      kind: suiTx.kind,
+      gasPayment: suiTx.gasPayment,
+      gasPrice: suiTx.gasPrice,
+      gasBudget: suiTx.gasBudget,
+      sender: suiTx.sender,
+    };
+
+    const dataBytes = bcs.ser('TransactionData', txData).toBytes();
+    const type_tag = Array.from('TransactionData::').map((e) => e.charCodeAt(0));
+    const serialized = new Uint8Array(type_tag.length + dataBytes.length);
+    serialized.set(type_tag);
+    serialized.set(dataBytes, type_tag.length);
+    if (this.signature.length > 0) {
+      const scheme = new Uint8Array([0x00]);
+      const signatureBuffer = Buffer.concat([
+        scheme,
+        Buffer.from(this._signature, 'hex'),
+        Buffer.from(this._publicKey, 'hex'),
+      ]);
+      const signedTx = {
+        data: txData,
+        txSignature: signatureBuffer,
+      };
+      const serialized = bcs.ser('SenderSignedData', signedTx).toBytes();
+      const typeTag = Array.from('SenderSignedData::').map((e) => e.charCodeAt(0));
+      const serializedFinal = new Uint8Array(typeTag.length + serialized.length);
+      serializedFinal.set(typeTag);
+      serializedFinal.set(serialized, typeTag.length);
+      this._id = Buffer.from(sha3.sha3_256(serializedFinal), 'hex').toString('base64');
+    }
+    return Buffer.from(serialized).toString('hex');
+  }
+
+  static deserializeSuiTransaction(serializedHex: string): SuiTransaction {
+    const data = Buffer.from(serializedHex, 'hex');
+    const trimmedData = new Uint8Array(data.subarray(TYPE_TAG.length));
```

Comment:
type TYPE_TAG is a global variable here

Reply:
yes, since it will always be "TransactionData::", I made it a global constant